plugins/wyze/src/main.py

```diff
@@ -542,6 +560,67 @@ async def _bootstrap_cruise_points(self):
         except Exception:
             pass
 
+    def _should_emit_ptz_caps(self, caps_json: str, force: bool = False) -> bool:
+        if not force and caps_json == self._last_ptz_caps_json:
+            return False
+        self._last_ptz_caps_json = caps_json
+        return True
+
+    async def _emit_ptz_caps_json(self, caps_json: str):
+        try:
+            await scrypted_sdk.deviceManager.onDeviceEvent(
+                self.nativeId, "ptzCapabilitiesJson", caps_json
+            )
+            await scrypted_sdk.deviceManager.onDeviceEvent(
+                self.nativeId, "wyze.ptzCapabilitiesJson", caps_json
+            )
+            try:
+                self.print("ptzCapabilities published:", caps_json)
+            except Exception:
+                pass
+        except Exception:
+            pass
+
+    async def _publish_ptz_presets(self, points: List[Dict[str, int]]):
+        if not self.camera.is_pan_cam:
+            return
+        try:
+            display_points = _dedupe_cruise_points(points)
+            presets = {}
+            for i, p in enumerate(display_points):
+                vd = p.get("vertical_degrees")
+                hd = p.get("horizontal_degrees")
+                if vd is None:
+                    vd = _vertical_to_degrees(p.get("vertical", 0))
+                if hd is None:
+                    hd = _horizontal_to_degrees(p.get("horizontal", 0))
+                label = f"Preset {i + 1}"
+                if vd is not None and hd is not None:
+                    label = f"Preset {i + 1} (tilt {vd}, pan {hd})"
+                presets[str(i + 1)] = label
+            caps = getattr(self, "ptzCapabilities", {"pan": True, "tilt": True})
+            caps = {**caps, "presets": presets}
+            caps_json = json.dumps(caps, default=str)
+            if not self._should_emit_ptz_caps(caps_json):
+                return
+            self.ptzCapabilities = caps
+            await scrypted_sdk.deviceManager.onDeviceEvent(
```

Comment:
You do not need to the event emit here, setting ptzCapabilities will do it automatically.